main.go

```diff
@@ -66,53 +72,114 @@ const (
 	webhookPort = 9443
 )
 
-func runPrometheusServer(metricsAddr string, tlsOptions common.SSPTLSOptions) error {
-	setupLog.Info("Starting Prometheus metrics endpoint server with TLS")
-	metrics.Registry.MustRegister(common_templates.CommonTemplatesRestored)
-	metrics.Registry.MustRegister(common.SSPOperatorReconcileSucceeded)
-	handler := promhttp.HandlerFor(metrics.Registry, promhttp.HandlerOpts{})
-	mux := http.NewServeMux()
-	mux.Handle("/metrics", handler)
+// This callback executes on each client call returning a new config to be used
+// please be aware that the APIServer is using http keepalive so this is going to
+// be executed only after a while for fresh connections and not on existing ones
+func getConfigForClientCallback(cfg *tls.Config, apiClient client.Client, ctx context.Context) (*tls.Config, error) {
+	var err error
+	var tlsProfile *ocpconfigv1.TLSSecurityProfile
 
-	minTlsVersion, err := tlsOptions.MinTLSVersionId()
-	if err != nil {
-		return err
+	var sspList ssp.SSPList
+	if err := apiClient.List(ctx, &sspList, &client.ListOptions{}); err != nil {
+		return nil, err
 	}
 
-	tlsConfig := tls.Config{
-		CipherSuites: tlsOptions.CipherIDs(&setupLog),
-		MinVersion:   minTlsVersion,
+	if len(sspList.Items) != 0 {
+		tlsProfile = sspList.Items[0].Spec.TLSSecurityProfile
 	}
 
-	server := http.Server{
-		Addr:      metricsAddr,
-		Handler:   mux,
-		TLSConfig: &tlsConfig,
+	if tlsProfile == nil {
+		cfg.MinVersion = crypto.DefaultTLSVersion()
+		cfg.CipherSuites = nil
+		return cfg, nil
+	}
+
+	if tlsProfile.Type == ocpconfigv1.TLSProfileCustomType {
+		cfg.CipherSuites = common.CipherIDs(tlsProfile.Custom.Ciphers)
+		cfg.MinVersion, err = crypto.TLSVersion(string(tlsProfile.Custom.MinTLSVersion))
+		if err != nil {
+			return nil, err
+		}
+		return cfg, nil
+	}
+
+	cipherNames, minTypedTLSVersion := ocpconfigv1.TLSProfiles[tlsProfile.Type].Ciphers, ocpconfigv1.TLSProfiles[tlsProfile.Type].MinTLSVersion
+	cfg.CipherSuites = common.CipherIDs(cipherNames)
+	cfg.MinVersion, err = crypto.TLSVersion(string(minTypedTLSVersion))
+	if err != nil {
+		return nil, err
 	}
+	return cfg, nil
+}
+
+type prometheusServer struct {
+	server   *http.Server
+	client   client.Client
+	certPath string
+	keyPath  string
+}
+
+func (s *prometheusServer) InjectClient(client client.Client) error {
```

Comment:
Make the `client` field public instead?

Reply:
The `controller-runtime` `Manager` uses this interface to inject client to any `Runnable`, when the client is created.

Reply:
Might add a comment about this interface?

Reply:
As discussed today, can you move this based on deprecation of the `InjectClient` interface to use `mgr.GetClient()` directly instead?